### PR TITLE
Feature/df 2609  add os business logs

### PR DESCRIPTION
**Pull request by emina-str**

### CODE REVIEW

File: application/src/test/java/ch/admin/seco/alv/service/onlineform/application/bu/ApplicationDocumentApplicationServiceTest.java [hunk 1]
```diff
@@ -123,6 +130,12 @@ public void testDeleteOldApplicationDocument() throws IOException {
 
         // then
         assertThat(this.applicationDocumentRepository.findAllByOwnerUserId(USER_1)).isEmpty();
+
```
Comment (seco-mgabor) [inline application/src/test/java/ch/admin/seco/alv/service/onlineform/application/bu/ApplicationDocumentApplicationServiceTest.java:133]: Can you please explain this code here?

File: application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java [hunk 2]
```diff
@@ -204,6 +212,14 @@ public void testRemoveWorkEffort() {
 
 		//then
 		assertThat(this.proofOfWorkEffortsApplicationService.getById(proofOfWorkEfforts.getId()).getWorkEfforts()).isEmpty();
+
```
Comment (seco-mgabor) [inline application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:215]: Why are you making assertions in the setUp()? This method is suppose to initialise stuff, not test it.

**Review by seco-mgabor — CHANGES_REQUESTED** [review]

**Comment by seco-mgabor** [comment]
Changes are OK, but the setUp() methods in test classes are very strange to me.

File: application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java (hunk 2 above)
Comment (emina-str) [reply application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:215]: Hey @seco-mgabor , on both comments you added, can you please just expand the file? the asserts are not in the setUp(), then on normal tests methods.

Comment (seco-mgabor) [reply application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:215]: My appologies. :) I take my comments back.

**Review by seco-mgabor — APPROVED** [review]

File: application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java [hunk 3]
```diff
@@ -332,6 +348,9 @@ public void testSubmitAllByNextSubmissionDate() {
 		isSubmitted(this.proofOfWorkEffortsRepository.findById(powe2019_05.getId()).get());
 		isSubmitted(this.proofOfWorkEffortsRepository.findById(powe2019_01.getId()).get());
 		isSubmitted(this.proofOfWorkEffortsRepository.findById(powe2018_12.getId()).get());
+
+		ArgumentCaptor<BusinessLogData> argumentCaptor = ArgumentCaptor.forClass(BusinessLogData.class);
+		verify(businessLogger, times(3)).log(argumentCaptor.capture());
```
Comment (palsch) [inline application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:353]: You argument captor is not used here, maybe it is better to replace that with any() or add additional asserts that uses the captor.

Comment (seco-mgabor) [reply application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:353]: I don't see the point of log() call too

Comment (emina-str) [reply application/src/test/java/ch/admin/seco/alv/service/onlineform/application/npa/ProofOfWorkEffortsApplicationServiceTest.java:353]: @palsch , sorry, I updated the test to check the even type.

**Pull request merged**